### PR TITLE
Edit drinks

### PR DESCRIPTION
**Pull request by csnedecor**
Adds ability to edit drinks

### CODE REVIEW

File: app/controllers/drinks_controller.rb [hunk 1]
```diff
@@ -22,6 +21,38 @@ def show
...
 
+  def edit
+    @drink = Drink.find(params[:id])
+    @liquors = Liquor.all
+    if !signed_in?
+      flash[:error] = "You must be signed in to do that"
+      redirect_to drink_path(@drink)
+    elsif @drink.user != current_user
+      flash[:error] = "You cannot edit someone else's drink"
+      redirect_to drink_path(@drink)
```
Comment (omidbachari) [inline app/controllers/drinks_controller.rb]: Is a `redirect_to` command (and a new HTTP request) the intended action here? Would it be better to keep the user in their existing client state with a `render`?

File: app/controllers/drinks_controller.rb [hunk 2]
```diff
@@ -22,6 +21,38 @@ def show
...
+      flash[:notice] = "Successfully updated drink"
+      redirect_to drink_path(drink)
+    else
+      @liquors = Liquor.all
+      if drink.alcoholic?
+        @drink = Drink.create(drink_params)
+      else
+        @drink = Drink.new(drink_params)
+        @drink.alcoholic = false
+        @drink.save
```
Comment (omidbachari) [inline app/controllers/drinks_controller.rb]: Is line 48 being committed to memory?

File: app/views/drinks/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  </div>
+  <% if @drink.alcoholic? %>
+    <p>Liquors</p>
+    <div>
+      <%= f.collection_check_boxes :liquor_ids,
+      @liquors,
+      :id,
+      :name
+      %>
+    </div>
```
Comment (omidbachari) [inline app/views/drinks/edit.html.erb]: Would it be appropriate to apprise the user who is editing a non-alcoholic drink that the drink is in fact non-alcoholic? Do you do that somewhere else?

File: db/migrate/20150102191126_add_default_avatar_url.rb [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
   def change
-    change_column
+    change_column(
       :users,
       :avatar,
       :string,
       default: "https://s3-us-west-2.amazonaws.com/tipsy-tally/default.jpg"
+      )
   end
 end
```
Comment (omidbachari) [inline db/migrate/20150102191126_add_default_avatar_url.rb]: I am assuming that `null: false` is already an existing db constraint.

File: app/controllers/drinks_controller.rb (hunk 1 above)
Comment (csnedecor) [inline app/controllers/drinks_controller.rb]: Seemed like it made sense to have them redirected to the path of the drink they're trying to edit so they can see that it is not their drink.

**Pull request merged**